contracts/Staking.sol

```diff
@@ -458,66 +446,110 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         totalStakedHistory.add64(getBlockNumber64(), newStake);
     }
 
-    /**
-     * Note: So far this function is called from unlock and from transferFromLock,
-     * which both ensure that lock can be unlocked (the latter through isLockManager modifier)
-     */
-    function _unlock(address _accountAddress, uint256 _lockId) internal {
+    function _unsafeUnlock(address _accountAddress, address _lockManager) internal {
         Account storage account = accounts[_accountAddress];
-        Lock storage lock_ = account.locks[_lockId];
+        Lock storage lock_ = account.locks[_lockManager];
 
-        lock_.unlockedAt = getTimestamp64();
+        // update total
+        // no need for SafeMath: totalLocked must be greater or equal than lock._amount
+        account.totalLocked = account.totalLocked - lock_.amount;
```

Comment:
We can follow the same rule we did for the court, using safemath unless it is checked in the same function